### PR TITLE
Scrimmage day code

### PR DESCRIPTION
**Pull request by 7Koris**
Code Used at Scrimmage Event should be added to master soon. We should tear it apart for any issues and ready it for Master. Contains an auto and teleop that worked on field, nonetheless we should still review.

### CODE REVIEW

File: src/main/java/frc5124/robot2020/RobotContainer.java [hunk 1]
```diff
@@ -168,6 +167,7 @@ public void disabledInit() {
    * @return the command to run in autonomous
    */
   public Command getAutonomousCommand() {
-    return null;
+    return new TargetShootAuto(shooter, loader, turret, driveTrain);
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/RobotContainer.java:176]: This is good

File: src/main/java/frc5124/robot2020/RobotContainer.java [hunk 2]
```diff
@@ -57,15 +60,17 @@
...
+  public JoystickButton operatorY = new JoystickButton(operator, 4);
+  public JoystickButton operatorLB = new JoystickButton(operator, 5);
+  public JoystickButton operatorRB = new JoystickButton(operator, 6);
+  public JoystickButton operatorLT = new JoystickButton(operator, 7);
+  public JoystickButton operatorRT = new JoystickButton(operator, 8);
+  public JoystickButton operatorBack = new JoystickButton(operator, 9);
+  public JoystickButton operatorStart = new JoystickButton(operator,10);
+  public JoystickButton operatorTest = new JoystickButton(operator, 9);
   public JoystickButton operatorStickLeft = new JoystickButton(operator, XboxController.Button.kStickLeft.value);
   public JoystickButton operatorStickRight = new JoystickButton(operator, XboxController.Button.kStickRight.value);
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/RobotContainer.java]: Why the change from Xbox enum to straight numbers? I don't have a preference, but I'm curious why.
Also, lines 74 and 75 should be consistent with the decision for the rest of the constructions.

File: src/main/java/frc5124/robot2020/RobotMap.java [hunk 3]
```diff
@@ -33,12 +32,13 @@
...
+        public static final double Kp = 0.000379; //,00021
         public static final double Ki = 0;
-        public static final double Kd = 0.00002;
-        public static final double Kf = 0.00008;
-        public static final double lineShootRPM = 4350; 
-        public static final double trenchShootRPM = 5200;
+        public static final double Kd = 0.005000;
+        public static final double Kf = 0.000040;
+        public static final double lineShootRPM = 4400; //4350
+        public static final double trenchShootRPM = 5300; //5200
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/RobotMap.java:50]: Why are we leaving the old values in constants? If it works better, can we just remove the old constants?

File: src/main/java/frc5124/robot2020/commands/auto/runpos/DriveForTime.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  public DriveForTime(DriveTrain subsystem, double time) {
+    super(time);
+    driveTrain = subsystem;
+    addRequirements(subsystem);
+  }
+
+  @Override
+  public void execute() {
+    super.execute();
+    driveTrain.tankDrive(.4, .4);
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/commands/auto/runpos/DriveForTime.java:27]: could we make the drive speed parametized instead of plugging it in inside the command? My assumption when I read "Drive For Time" is that it will be full power (which is agreeably ridiculous). I think that the power choice should be made within the auto group itself.

File: src/main/java/frc5124/robot2020/commands/auto/runpos/ShooterAndLoaderRev.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  }
+   
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    if (!startedShooter) {
+      m_shooter.startShooter();
+      startedShooter = true;
+    }
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/commands/auto/runpos/ShooterAndLoaderRev.java:37]: Use the initialize method for this

File: src/main/java/frc5124/robot2020/commands/auto/runpos/Turn180.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc5124.robot2020.commands.auto.runpos;
+
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc5124.robot2020.subsystems.Turret;
+import edu.wpi.first.networktables.NetworkTableInstance;
+
+public class Turn180 extends WaitCommand {
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/commands/auto/runpos/Turn180.java:14]: The command is called turn 180 but it turns to the position -180. That's kind of confusing: consider a rename.
Also, instead of building the timeout into the command, timeouts can be specified as a decoration to the command.

File: src/main/java/frc5124/robot2020/commands/auto/runpos/turretZero.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+package frc5124.robot2020.commands.auto.runpos;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc5124.robot2020.subsystems.Turret;
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/latest/docs/software/commandbased/convenience-features.html
+public class turretZero extends InstantCommand {
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/commands/auto/runpos/turretZero.java]: You can use a lambda to do this in an instant command instead of overriding initialize. This unnecessarily couples our command to the specifics of how an instant command works, instead of interfacing to the command and being agnostic of its internals. Alternatively, if lambdas are taboo, it would be better just to extend CommandBase.

File: src/main/java/frc5124/robot2020/commands/loader/SeeBallRunBelt.java [hunk 8]
```diff
@@ -14,7 +14,7 @@
 
   private Loader m_Loader;
 
-  public SeeBallRunBelt(Loader subsystem) {
+  public  SeeBallRunBelt(Loader subsystem) {
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/commands/loader/SeeBallRunBelt.java:17]: what is this? put it back

File: src/main/java/frc5124/robot2020/commands/turret/RotateTurret.java [hunk 9]
```diff
@@ -40,12 +41,16 @@ public void initialize() {
...
     } else {
       turret.directPower(0);
     }
+
+    
+
+    SmartDashboard.putNumber("TurretDegrees", turret.getDegrees());
   }
 
   // Called once the command ends or is interrupted.
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/commands/turret/RotateTurret.java:50]: This shouldn't be a single command that accepts buttons and uses them. It should be a pair of while held commands, one for each direction.

File: src/main/java/frc5124/robot2020/subsystems/DriveTrain.java [hunk 10]
```diff
@@ -50,6 +52,11 @@ public DriveTrain() {
         rightFollower = new WPI_TalonFX(RobotMap.DriveTrainMap.rightFollowerCanID);
         rightFollower.follow(rightLeader);
 
+        rightLeader.setNeutralMode(NeutralMode.Brake);
+        rightFollower.setNeutralMode(NeutralMode.Brake);
+        leftLeader.setNeutralMode(NeutralMode.Brake);
+        leftFollower.setNeutralMode(NeutralMode.Brake);
+
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/subsystems/DriveTrain.java]: Does Will no longer want it to be half and half, or is this just the wrong code version?

File: src/main/java/frc5124/robot2020/subsystems/DriveTrain.java [hunk 11]
```diff
@@ -101,7 +108,7 @@ public void periodic() {
     // Control methods
 
     public void tankDrive(double left, double right) {
-        differentialDrive.tankDrive(left,right);   
+        differentialDrive.tankDrive(left,-right);
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/subsystems/DriveTrain.java]: Differential drive should take care of the negation: are you sure this is right? We should be able to get this to work using the motor directions or something.

File: src/main/java/frc5124/robot2020/subsystems/Hanger.java [hunk 12]
```diff
@@ -19,15 +19,17 @@
 
 public class Hanger implements Subsystem {
   private CANSparkMax hangerMotor;
-  private Solenoid brake;
+  private Solenoid noBrake;
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/subsystems/Hanger.java:25]: How about naming it "brakeReleaser" or "unbraker"? This commit change looks like manufacturing decided to take the brake off the robot so we the programmers programmed it anyways or smth.

File: src/main/java/frc5124/robot2020/subsystems/Hanger.java [hunk 13]
```diff
@@ -64,8 +72,9 @@ public boolean reachedBottomLimit(){
     return bottomLimit.get();
   }
 
+
   public void setNoPower(){
     hangerMotor.set(RobotMap.HangerMap.hangerHalt);
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/subsystems/Hanger.java:85]: why not just say "zero" instead of hanger halt

File: src/main/java/frc5124/robot2020/subsystems/Intake.java [hunk 14]
```diff
@@ -36,6 +39,11 @@ public boolean isDeployed() {
 
   public void setIntakePower(double power){
     rollerSpeedController.set(power);
+    if (power > 0 || power < 0) {
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/subsystems/Intake.java]: use power != 0, or better, Math.abs(power) > 0.01

File: src/main/java/frc5124/robot2020/subsystems/Loader.java [hunk 15]
```diff
@@ -35,10 +35,13 @@ public void setPower(double power){
...
   }
+
+  public double getAppliedOutput() {
+    return topBeltMotor.getAppliedOutput();
+  }
   
   public void runBelt() {
-    topBeltMotor.set(RobotMap.LoaderMap.beltSpeed);
-    //bottomBeltMotor.set(.2);
+    topBeltMotor.set(.420);
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/subsystems/Loader.java]: Lets put it back to using the constant, and set the constant to .42 instead of doing it here.
Sorry, this one's my bad.

File: src/main/java/frc5124/robot2020/subsystems/Turret.java [hunk 16]
```diff
@@ -80,6 +74,13 @@ public void isManual(boolean manual) {
     this.manual = manual;
   }
 
+  public void turretLimitSet() {
+    turretMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -54);
+    turretMotor.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 5);
```
Comment (Randall-Scharpf) [inline src/main/java/frc5124/robot2020/subsystems/Turret.java]: use a number of rotations or degrees and a conversion, I don't speak Kam units

File: src/main/java/frc5124/robot2020/RobotContainer.java (hunk 2 above)
Comment (SrilakshmiC) [reply src/main/java/frc5124/robot2020/RobotContainer.java]: i changed the controller mode (there's a little switch on the back of the controller). It basically makes the triggers as buttons instead of axes. with that, the values of these buttons changed, so I just set straight numbers. I agree w the changing lines 74 & 75. I believe those are buttons 11 (left) & 12 (right).

File: src/main/java/frc5124/robot2020/RobotMap.java (hunk 3 above)
Comment (SrilakshmiC) [reply src/main/java/frc5124/robot2020/RobotMap.java:50]: I'd want to keep those old values in case of anything. It doesn't really hurt to have them. But, if you want to delete them, I'm okay with that too. My opinion on that is just kinda like we should try to do that so that if we have to change it, we have a baseline to change it from. But, these numbers aren't super different so, in this case I don't think it matters much.

File: src/main/java/frc5124/robot2020/commands/auto/runpos/DriveForTime.java (hunk 4 above)
Comment (SrilakshmiC) [reply src/main/java/frc5124/robot2020/commands/auto/runpos/DriveForTime.java:27]: agreed

File: src/main/java/frc5124/robot2020/subsystems/DriveTrain.java (hunk 10 above)
Comment (7Koris) [reply src/main/java/frc5124/robot2020/subsystems/DriveTrain.java]: wrong ver

File: src/main/java/frc5124/robot2020/RobotContainer.java (hunk 2 above)
Comment (theideman) [reply src/main/java/frc5124/robot2020/RobotContainer.java]: I agree with Randall, should continue using constants. You also have 2 button variables assigned to the same button (9).

File: src/main/java/frc5124/robot2020/RobotContainer.java [hunk 17]
```diff
@@ -98,24 +103,25 @@ private void configureSubsystems() {
...
     operatorStart.whileHeld(new SetIntakePower(intake, -.6));
-    operatorBack.whileHeld(new ReverseBeltWithIntake(loader, intake));
+    operatorBack.whileHeld(new ReverseBeltWithIntakeAndShooter(shooter, loader, intake));
     operatorX.whileHeld(new LoaderAndIntakeGroup(intake, loader));
-    operatorA.whenPressed(new ToggleIntakePivot(intake));
-    operatorB.toggleWhenPressed(new RotateTurret(turret, operatorRight, operatorLeft)).whenInactive(new TurretTargetByPIDPerpetually(turret));
-    
-    //operatorDown.whileHeld(new LiftDown(hanger));
-    operatorY.whileHeld(new RunLoader(loader));
+    //AAA???
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/RobotContainer.java:107]: remove this

File: src/main/java/frc5124/robot2020/RobotContainer.java [hunk 18]
```diff
@@ -98,24 +103,25 @@ private void configureSubsystems() {
...
-    //operatorRB.whileHeld(new RotateTurret(turret, RobotMap.TurretMap.turretSpeed));
-   //operatorLB.whileHeld(new RotateTurret(turret, -RobotMap.TurretMap.turretSpeed));     
-    
+   
   }
 
   private void configureDefaultCommands(){
     driveTrain.setDefaultCommand(new JoystickTankDrive(driverLeft, driverRight, driveTrain));
-    //turret.setDefaultCommand(new TurretTargetByPIDPerpetually(turret));
+    turret.setDefaultCommand(new TurretFindHome(turret));
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/RobotContainer.java:122]: why is this a default command? it can be executed at initialization if necessary and should have a button for the operator to reset to home on command

File: src/main/java/frc5124/robot2020/RobotContainer.java [hunk 19]
```diff
@@ -168,6 +174,7 @@ public void disabledInit() {
    * @return the command to run in autonomous
    */
   public Command getAutonomousCommand() {
-    return null;
+    return new TargetShootAuto(shooter, loader, turret, driveTrain);
+    //TargetShootAuto(shooter, loader, turret, driveTrain);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/RobotContainer.java:177]: remove commented code

File: src/main/java/frc5124/robot2020/RobotMap.java (hunk 3 above)
Comment (theideman) [reply src/main/java/frc5124/robot2020/RobotMap.java:50]: agree they should be removed. the point of source code control is that we can look back at changes. if these values were never good enough to be committed to master, then there's no point in keeping them around

File: src/main/java/frc5124/robot2020/commands/auto/runpos/DriveForTime.java (hunk 4 above)
Comment (theideman) [reply src/main/java/frc5124/robot2020/commands/auto/runpos/DriveForTime.java:27]: drive by time should not be something we do in the final code. we need to get to running by encoders for distance.

Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/auto/runpos/DriveForTime.java:27]: Speed/power values should be constants.

File: src/main/java/frc5124/robot2020/commands/auto/runpos/Turn180.java [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+  private Turret turret;
+  private boolean setTarget = false;
+  /**
+   * Creates a new DriveForTime.
+   */
+  public Turn180(Turret subsystem) {
+    super(5); //5 -> 3?
+    turret = subsystem;
+    addRequirements(subsystem);
+    NetworkTableInstance.getDefault().getTable("limelight").getEntry("pipeline").setDouble(0.0);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/auto/runpos/Turn180.java:24]: might want to make a class to get common values from the network table

File: src/main/java/frc5124/robot2020/commands/driveTrain/JoystickTankDrive.java [hunk 21]
```diff
@@ -30,10 +30,11 @@ public JoystickTankDrive(Joystick leftHand, Joystick rightHand, DriveTrain subsy
     public void execute() {
 
         if (leftHand.getY() > 0.1 || rightHand.getY() > 0.1 || rightHand.getY() < -0.1 || leftHand.getY() < -0.1){
-            leftHandIn = Math.pow(leftHand.getY(), 3);
-            rightHandIn = Math.pow(rightHand.getY(), 3);
-            m_driveTrain.tankDrive(leftHandIn, rightHandIn);
+            leftHandIn = Math.pow(-leftHand.getY(), 1);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/driveTrain/JoystickTankDrive.java:33]: does this raise the Y value by the power of 1? is that necessary?

File: src/main/java/frc5124/robot2020/commands/driveTrain/JoystickTankDrive.java [hunk 22]
```diff
@@ -30,10 +30,11 @@ public JoystickTankDrive(Joystick leftHand, Joystick rightHand, DriveTrain subsy
...
+           
         }
+       rightHandIn = Math.pow(-rightHand.getY(), 1);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/driveTrain/JoystickTankDrive.java:36]: same as above

File: src/main/java/frc5124/robot2020/commands/shooter/ReverseShooter.java [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    m_shooter.directPower(-.3);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/shooter/ReverseShooter.java:35]: use constant

File: src/main/java/frc5124/robot2020/commands/shooter/ShootFromMidTrench.java [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+  public void initialize() {
+    m_shooter.startShooter(RobotMap.ShooterMap.midTrenchShootRPM);
+    SmartDashboard.putBoolean("ShooterRunning", true);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() { 
+    // m_shooter.currentWatch(RobotMap.ShooterMap.lineShootRPM);
+    if (m_shooter.getVelocity() >= RobotMap.ShooterMap.midTrenchShootRPM-20 && m_loader.getAppliedOutput() == 0) {
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/shooter/ShootFromMidTrench.java:42]: this line still only checks if we are below the shooting threshold, but does not consider if we are spiking above the threshold

Comment (theideman) [reply src/main/java/frc5124/robot2020/commands/shooter/ShootFromMidTrench.java:42]: I suggest using the logic math.abs(getVelocity()-shootRPM) <= 20 to capture the range above and below the set point

Comment (theideman) [reply src/main/java/frc5124/robot2020/commands/shooter/ShootFromMidTrench.java:42]: I see this logic is copied in other places. consider adding a method in the shooter subsystem that can give a boolean response when we have a valid firing solution

File: src/main/java/frc5124/robot2020/commands/shooter/ShootFromTrench.java [hunk 25]
```diff
@@ -16,34 +16,34 @@
...
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void execute() { 
-    // m_shooter.currentWatch(RobotMap.ShooterMap.lineShootRPM);
-    //if (m_shooter.getVelocity() >= RobotMap.ShooterMap.trenchShootRPM-15 && m_loader.getAppliedOutput() == 0) {
-    //   m_loader.runBelt();
-    // } 
+     m_shooter.currentWatch(RobotMap.ShooterMap.lineShootRPM);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/shooter/ShootFromTrench.java:41]: duplicate logic here

File: src/main/java/frc5124/robot2020/commands/turret/RotateTurret.java [hunk 26]
```diff
@@ -12,40 +12,39 @@
...
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   // Need encoder position limits to finish coding
   @Override
   public void execute() {
-    if (operatorRight.get()) {
-      turret.directPower(.2);
-    }else if (operatorLeft.get()) {
+    if (clockwise && !turret.isAutomatic()) {
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/turret/RotateTurret.java:39]: constant

File: src/main/java/frc5124/robot2020/commands/turret/TurretDefaultBehaviour.java [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+  public TurretDefaultBehaviour(Turret subsystem) {
+    turret = subsystem;
+    addRequirements(turret);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    if (!(turret.setHome())) {
+     turret.directPower(.1);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/turret/TurretDefaultBehaviour.java:38]: constant

File: src/main/java/frc5124/robot2020/commands/turret/TurretDefaultBehaviour.java [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+     SmartDashboard.putNumber("yeet", 0);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/turret/TurretDefaultBehaviour.java:39]: I wouldn't know what this means on the dashboard

File: src/main/java/frc5124/robot2020/commands/turret/TurretFindHome.java [hunk 29]
```diff
@@ -0,0 +1,108 @@
...
+  public TurretFindHome(Turret subsystem) {
+    turret = subsystem;
+    addRequirements(turret);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    if (!(turret.setHome()) && !turret.initialHome()) {
+     turret.directPower(.1);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/turret/TurretFindHome.java:36]: constant

File: src/main/java/frc5124/robot2020/commands/turret/TurretFindHome.java [hunk 30]
```diff
@@ -0,0 +1,108 @@
...
+     SmartDashboard.putNumber("yeet", 0);
+     SmartDashboard.updateValues();
+  }
+
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/turret/TurretFindHome.java:45]: clean up indentation and commented code

File: src/main/java/frc5124/robot2020/commands/turret/TurretTargetByPID.java [hunk 31]
```diff
@@ -18,7 +18,6 @@
    */
   public TurretTargetByPID(Turret subsystem) {
     this.subsystem = subsystem;
-    addRequirements(subsystem);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/turret/TurretTargetByPID.java:21]: why was this removed? should be necessary.

File: src/main/java/frc5124/robot2020/commands/turret/setTurretDegrees.java [hunk 32]
```diff
@@ -18,7 +18,6 @@
    */
   public setTurretDegrees(Turret subsystem, double degrees) {
     turret = subsystem;
-    addRequirements(turret);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/commands/turret/setTurretDegrees.java:21]: why is this removed?

File: src/main/java/frc5124/robot2020/subsystems/Camera.java [hunk 33]
```diff
@@ -8,14 +8,20 @@
...
 public class Camera implements Subsystem {
+  private NetworkTableEntry shuffleboardButtonBooleanEntry;
+  private ShuffleboardTab display;
   
   public Camera() {
   }
 
   @Override
   public void periodic() {
+    if (RobotMap.debugEnabled) {}
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/subsystems/Camera.java:24]: what is the purpose of always checking if debug enabled and then doing nothing in response?

File: src/main/java/frc5124/robot2020/subsystems/DriveTrain.java [hunk 34]
```diff
@@ -80,6 +91,7 @@ public DriveTrain() {
 
     @Override
     public void periodic() {
+        if (RobotMap.debugEnabled) {}
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/subsystems/DriveTrain.java:94]: what is this for? seems unnecessary

File: src/main/java/frc5124/robot2020/subsystems/DriveTrain.java [hunk 35]
```diff
@@ -101,7 +113,7 @@ public void periodic() {
     // Control methods
 
     public void tankDrive(double left, double right) {
-        differentialDrive.tankDrive(left,right);   
+        differentialDrive.tankDrive(left, -right);
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/subsystems/DriveTrain.java:116]: should be inverting motors rather than passing in negated inputs

File: src/main/java/frc5124/robot2020/subsystems/Hanger.java [hunk 36]
```diff
@@ -16,44 +16,60 @@
...
 import edu.wpi.first.wpilibj2.command.Subsystem;
 import frc5124.robot2020.RobotMap;  
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
 
 public class Hanger implements Subsystem {
   private CANSparkMax hangerMotor;
-  private Solenoid brake;
+  private Solenoid noBrake;
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/subsystems/Hanger.java:25]: seems odd to use the negative here, would be clearer to read code if brake is on or off rather than the no brake is true or false.

File: src/main/java/frc5124/robot2020/subsystems/Hanger.java [hunk 37]
```diff
@@ -16,44 +16,60 @@
...
    
   public void liftUp() {
-    if (reachedTopLimit()) {
-      setNoPower();
-    }
-    else {
-      brake.set(false);
-      hangerMotor.set(RobotMap.HangerMap.hangerMotor);
-    }
+    // if (reachedTopLimit()) {
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/subsystems/Hanger.java:50]: clean up commented code

File: src/main/java/frc5124/robot2020/subsystems/Intake.java [hunk 38]
```diff
@@ -4,29 +4,40 @@
...
       rollerSpeedController.setInverted(false);
       deployed = false;         // pivot is up 
+      SmartDashboard.putBoolean("IntakeRunning", false);
+      display = Shuffleboard.getTab("Intake Display");
+      Shuffleboard.update();
   }
 
   @Override
   public void periodic() {
+    if (RobotMap.debugEnabled) {}
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/subsystems/Intake.java:34]: again what is this?

File: src/main/java/frc5124/robot2020/subsystems/Turret.java [hunk 39]
```diff
@@ -99,6 +95,22 @@ public void disableTurretPID () {
...
 
+  public void setHome(boolean isHome) {
+    this.isHome = isHome;
+  }
+
+  public boolean setHome () {
+    return isHome;
+  }
+
+  public void initialHome(boolean initialHome) {
```
Comment (theideman) [inline src/main/java/frc5124/robot2020/subsystems/Turret.java:106]: getters and setters are more clearly identified when we use get and set in their names
